wallet/src/Pos/Wallet/Web/Methods/Payment.hs

```diff
@@ -57,6 +59,21 @@ newPayment passphrase srcAccount dstAddress coin policy =
         (one (dstAddress, coin))
         policy
 
+newPaymentBatch
+    :: MonadWalletTxFull ctx m
+    => PassPhrase
+    -> NewBatchPayment
+    -> m CTx
+newPaymentBatch passphrase NewBatchPayment {..} = do
+    src <- decodeCTypeOrFail npbFrom
+    dst <- nonEmpty npbTo `whenNothing`
+        throwM (RequestError "At least one recipient required.")
```

Comment:
you can use `NonEmpty` for `npbTo`

Reply:
yes, but then the same logic 
```
nonEmpty npbTo `whenNothing`
       throwM (RequestError "At least one recipient required.")
```

would have to be pushed inside `FromJSON` instance again (because there is no `FromJSON NonEmpty` and I would not use it as fail message is much more clear this way - mentioning recipients). 

On a second though I can add `FromJSON NonEmpty` and use generic fail message "Expected at least one element".

Do you think this would be better ?

Reply:
I think it would be better to check inside `FromJSON NewPaymentBatch`. it would just guarantee that you have at least one element and you don't need it check somewhere else. 
however, IMHO, it's more or less minor, I don't mind your approach.